packages/vuetify/src/components/VDivider/VDivider.ts

```diff
@@ -1,42 +1,43 @@
-// @ts-nocheck
-/* eslint-disable */
-
 // Styles
 import './VDivider.sass'
 
-// Types
-import { VNode } from 'vue'
+// Utilities
+import { computed, defineComponent, h } from 'vue'
 
-// Mixins
-import Themeable from '../../mixins/themeable'
+// Composables
+import { useTheme } from '@/composables'
 
-export default Themeable.extend({
-  name: 'v-divider',
+export default defineComponent({
+  name: 'VDivider',
 
   props: {
     inset: Boolean,
     vertical: Boolean,
   },
 
-  render (h): VNode {
-    // WAI-ARIA attributes
-    let orientation
-    if (!this.$attrs.role || this.$attrs.role === 'separator') {
-      orientation = this.vertical ? 'vertical' : 'horizontal'
-    }
-    return h('hr', {
-      class: {
-        'v-divider': true,
-        'v-divider--inset': this.inset,
-        'v-divider--vertical': this.vertical,
-        ...this.themeClasses,
-      },
-      attrs: {
-        role: 'separator',
-        'aria-orientation': orientation,
-        ...this.$attrs,
-      },
-      on: this.$listeners,
+  setup (props, { attrs }) {
+    const { themeClasses } = useTheme()
+    const classes = computed(() => {
+      return {
+        'v-divider--inset': props.inset,
+        'v-divider--vertical': props.vertical,
+      }
     })
+    const role = attrs.role ?? 'separator'
```

Comment:
This will not be reactive

Reply:
Tested this locally and it works. What am I missing?

```vue
<template>
  <v-app>
    <div class="ma-12 pa-12">
      <button @click="onClick">click me</button>

      <div class="my-5"></div>

      <v-divider :role="role"></v-divider>
    </div>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
      role: undefined,
    }),

    methods: {
      onClick () {
        this.role = this.role === undefined ? 'button' : undefined
      },
    },
  }
</script>
```